src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1348,7 +1348,9 @@ let genExpr (e: Expr) =
 
                 let longExpression =
                     genExpr node.FunctionExpr
-                    +> indentSepNlnUnindent (col sepNln node.Arguments genExpr)
```

Comment:
I actually refactored this code a while ago so you could add a new active pattern for this instead. Something very similar to `(|EndsWithSingleListApp|_|)` but it looks if the last expression is (anon) record and the new setting is on.